widget/richtext.go

```diff
@@ -549,14 +549,47 @@ func (r *textRenderer) layoutRow(texts []fyne.CanvasObject, align fyne.TextAlign
 		return texts[0].MinSize().Height
 	}
 	height := float32(0)
+	tallestBaseline := float32(0)
+	realign := false
 	for _, text := range texts {
-		size := text.MinSize()
-
+		var size fyne.Size
+		if txt, ok := text.(*canvas.Text); ok {
+			s, base := fyne.CurrentApp().Driver().RenderedTextSize(txt.Text, txt.TextSize, txt.TextStyle)
+			if tallestBaseline == 0 {
+				tallestBaseline = base
+			} else if base != tallestBaseline {
+				tallestBaseline = fyne.Max(tallestBaseline, base)
+				realign = true
+			}
+			size = s
+		} else {
+			size = text.MinSize()
+		}
 		text.Resize(size)
-		text.Move(fyne.NewPos(xPos, yPos)) // TODO also baseline align for height (need new measure info)
+		text.Move(fyne.NewPos(xPos, yPos))
+
 		xPos += size.Width
-		height = fyne.Max(height, size.Height)
+		if height == 0 {
+			height = size.Height
+		} else if height != size.Height {
+			height = fyne.Max(height, size.Height)
+			realign = true
+		}
 	}
+
+	if realign {
+		for _, text := range texts {
+			if _, ok := text.(*canvas.Text); !ok {
+				continue
+			}
+
+			txt := text.(*canvas.Text)
+			_, base := fyne.CurrentApp().Driver().RenderedTextSize(txt.Text, txt.TextSize, txt.TextStyle)
```

Comment:
Hmm I wonder if this has performance impact as `RenderedTextSize` is measuring the text each time even we only need the baseline, maybe the text baseline measurement should have its own function separated from the text size?

Reply:
The expensive part of measuring is the text lookup, parse and so forth.
The best solution is probably to cache text measurement, which I was going to add later.

Reply:
Yes, cache could help, however I think it would be great, if apart from the RenderedTextSize, an additional baseline function separated from this, because as far as I know to get the baseline we just need the text style, text size, and one random character 🧐

Reply:
The cache is in. I would like to avoid adding new methods if possible. Should we need to expose it later we can - but probably by replacing `MeasureText` with `TextMetrics` instead of adding another `TextBaseline` and how many others may be added in the future

Reply:
Sounds good